lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBODoiDaoImpl.java

```diff
@@ -104,60 +89,68 @@ public Doi createDoi(final String userGroupId, final String objectId,
 	 */
 	@NewWriteTransaction
 	@Override
-	public Doi updateDoiStatus(final String objectId, final ObjectType objectType,
-			final Long versionNumber, final DoiStatus doiStatus, String etag)
+	public Doi updateDoiStatus(Doi dto)
 			throws NotFoundException, DatastoreException, ConflictingUpdateException {
 
-		if (objectId == null || objectId.isEmpty()) {
+		if (dto.getObjectId() == null) {
 			throw new IllegalArgumentException("Object ID cannot be null nor empty.");
 		}
-		if (objectType == null) {
+		if (dto.getObjectType() == null) {
 			throw new IllegalArgumentException("Object type cannot be null.");
 		}
-		if (doiStatus == null) {
+		if (dto.getDoiStatus() == null) {
 			throw new IllegalArgumentException("DOI status cannot be null.");
 		}
+		DBODoi dbo = getDbo(dto);
 
-		DBODoi dbo = getDbo(objectId, objectType, versionNumber);
-		if (!dbo.getETag().equals(etag)) {
+		if (!dbo.getETag().equals(dto.getEtag())) {
```

Comment:
move etag check to manager because

Add to the dao
String getEtagForUpdate(objectId, objectType, versionNumber) // select etag for update.